cnd/src/network.rs

```diff
@@ -54,7 +57,18 @@ impl Swarm {
         let local_peer_id = PeerId::from(local_key_pair.public());
         tracing::info!("Starting with peer_id: {}", local_peer_id);
 
-        let transport = transport::build_comit_transport(local_key_pair)?;
+        let transport = if settings.network.use_tor {
+            let mut map = HashMap::new();
+            if settings.network.listen.len() != 1 {
+                anyhow::bail!("To use Tor please configure a single onion address")
+            }
+            for addr in settings.network.listen.iter() {
+                map.insert((*addr).clone(), 9939u16);
```

Comment:
Does this mean we are gonna listen on 9939 locally and Tor has to be configured to route traffic do this port?

How does Tor know about this?

Reply:
Yes. That is configured in the torrc.
```
HiddenServicePort 9939 127.0.0.1:9939
```

The first port number is the virtual port (the port used as part of the onion address).

Previously I did this using the Tor control protocol but this is, in my opinion, a better way to do it.